File: chinese-shadowing-client/src/models/Section.js
import Persona from './Persona';

export default class Section {
  /** @type {string} */
  id;
  /** @type {number} */
  start;
  /** @type {number} */
  end;
  /** @type {string} */
  audioUrl;
  /** @type {string} */
  text;
  /** @type {Persona} */
  persona;

  constructor(section) {
    this.id = section.id;
    this.start = section.start;
    this.end = section.end;
    this.audioUrl = section.audioUrl;
    this.text = section.text;
    this.persona = new Persona(section.persona);
  }
}

File: chinese-shadowing-api/Shadowing.DataAccess/Entities/Section.cs
using System;

namespace Shadowing.DataAccess.Entities
{
    public class Section
    {
        public string Id { get; set; }
        public double Start { get; set; }
        public double End { get; set; }
        public string AudioUrl { get; set; }
        public string Text { get; set; }
        public string PersonaId { get; set; }
        public Persona Persona { get; set; }
        public string EpisodeId { get; set; }
        public Episode Episode { get; set; }
        public DateTime Created { get; set; }
        public DateTime? Updated { get; set; }
    }
}
